### PR TITLE
Skill - Petrify

### PR DESCRIPTION
**Pull request by JesterWizard**
A skill % chance of inflicting the petrify status on the opponent.

### CODE REVIEW

File: Wizardry/Core/BattleSys/Source/BattleHit.c [hunk 1]
```diff
@@ -145,6 +145,17 @@ void BattleGenerateHitAttributes(struct BattleUnit * attacker, struct BattleUnit
         gBattleStats.damage = gBattleStats.damage * 3 / 2;
 #endif
 
+#if defined(SID_Petrify) && (SID_Petrify < MAX_SKILL_NUM)
+    if (SkillTester(unit, SID_Petrify))
+    {
+        //Roll a random number against the attacker's skill
```
Comment (MokhaLeee) [inline Wizardry/Core/BattleSys/Source/BattleHit.c:151]: add a space after "//"

File: Wizardry/Core/BattleSys/Source/BattleHit.c [hunk 2]
```diff
@@ -145,6 +145,17 @@ void BattleGenerateHitAttributes(struct BattleUnit * attacker, struct BattleUnit
...
+        //Then check if the defending unit has no status
+        if(BattleRoll1RN(attacker->unit.skl, FALSE) == TRUE && GetUnitStatusIndex(&defender->unit) == UNIT_STATUS_NONE)
```
Comment (MokhaLeee) [inline Wizardry/Core/BattleSys/Source/BattleHit.c:153]: add space after "if"

Comment (JesterWizard) [reply Wizardry/Core/BattleSys/Source/BattleHit.c]: Done.

File: Wizardry/Core/BattleSys/Source/BattleHit.c (hunk 1 above)
Comment (JesterWizard) [reply Wizardry/Core/BattleSys/Source/BattleHit.c]: Done.

**Comment by MokhaLeee** [comment]
The activated skill should not put at pre-battle calc.
It should be rolled RN after MISS RN has been roled in `BattleGenerateHitAttributes`.
Kernel also offered one common roll RN API as `CheckBattleSkillActivte`

This PR will be rejected as because it fails to follow the function logic of the vanilla. It is recommended that you modify the code and then resubmit

**Comment by MokhaLeee** [comment]
PoisonPoint skill is also fixed

https://github.com/MokhaLeee/fe8u-cskillsys-kernel/pull/57

**Pull request closed without merging**